File: Stacks/stack_ll.c
#include<stdio.h>
#include<stdlib.h>
#include<limits.h>

struct StackNode{
	int data;
	struct StackNode* next;
};

struct StackNode* newNode(int data){
	struct StackNode* node = (struct StackNode*)malloc(sizeof(struct StackNode));
	node -> data = data;
	node -> next = NULL;
	return node;
}
void push(struct StackNode** root, int value){
	struct StackNode* stacknode = newNode(value);
	stacknode->next = *root;
	*root = stacknode;
	printf("%d pushed into stack \n", value );	
}
int isEmpty(struct StackNode* root){
	return !root;
}
void pop(struct StackNode** root){
	if(isEmpty(*root)){
		printf("Stack is Empty \n");
		return;
	}
	struct StackNode* temp = *root;
	*root = (*root) -> next;
	printf("%d popped from stack \n", temp->data);
	free(temp); 
}
int main()
{
	struct StackNode* root = NULL;
	push(&root , 1);
	push(&root , 2);
	push(&root , 3);

	pop(&root);
	pop(&root);
	pop(&root);
	pop(&root);
	return 0;
}

File: Stacks/stack_ll.py
class StackNode: 
	def __init__(self, data):
		self.data = data
		self.next = None

class Stack:
	def __init__(self):
		self.root = None

	def isEmpty(self):
		return True if self.root is None else False

	def push(self, data):
		newNode = StackNode(data)
		newNode.next = self.root
		self.root = newNode
		print( str(data) + " pushed into the stack")

	def pop(self):
		if self.isEmpty():
			print("Stack is Empty")
			return
		temp = self.root
		self.root = self.root.next
		print(str(temp.data) + " popped from the stack")

			
stack = Stack()
stack.push(1)
stack.push(2)
stack.push(3)

stack.pop()
stack.pop()
stack.pop()
stack.pop()